src/command.ts

```diff
@@ -1,97 +1,99 @@
 import {
   BotCommand,
   BotCommandScope,
+  BotCommandScopeAllChatAdministrators,
+  BotCommandScopeAllGroupChats,
+  BotCommandScopeAllPrivateChats,
   Chat,
   ChatTypeContext,
-  CommandMiddleware,
   Composer,
   Context,
   match,
   Middleware,
+  MiddlewareObj,
   P,
 } from "./deps.deno.ts";
 
 export type MaybeArray<T> = T | T[];
+export type ChatTypeMiddleware<C extends Context, T extends Chat["type"]> = Middleware<ChatTypeContext<C, T>>;
+type BotCommandGroupsScope = BotCommandScopeAllGroupChats | BotCommandScopeAllChatAdministrators;
 
-export class Command<C extends Context = Context> extends Composer<C> {
-  #scopes: BotCommandScope[] = [];
-  #languages: Map<string, { name: string; description: string }> = new Map();
+const isAdmin = (ctx: Context) =>
+  ctx.getAuthor().then((author) => ["administrator", "creator"].includes(author.status));
 
-  constructor(name: string, description: string, ...middleware: Array<Middleware<C>>) {
-    super();
+export class Command<C extends Context = Context> implements MiddlewareObj<C> {
+  private _scopes: BotCommandScope[] = [];
+  private _languages: Map<string, { name: string; description: string }> = new Map();
+  private _composer: Composer<C> = new Composer<C>();
```

Comment:
nit:
```suggestion
  private languages = new Map<string, { name: string; description: string }>();
  private composer = new Composer<C>();
```
I see no reason to prefix fields with `_` if they are already declared as private anyway (but this is merely a matter of personal taste).

Reply:
I need to expose `languages` in a getter, so it needs the `_`, and I kept it in the others for consistency and because I kind of like it.

Reply:
Why use a getter over a `readonly` property?

Reply:
We're nitpicking, it's irrelevant

Reply:
We are 😅
I’ll mark as resolved